Assignment-11-September-4/vite-project/src/Components/Dashboard/SurveyComponent.tsx

```diff
@@ -0,0 +1,97 @@
+import { useState } from "react";
+import { Box, Button, LinearProgress } from "@mui/material";
+import Question1 from "./Questions/Question1";
+import Question2 from "./Questions/Question2";
+import Question3 from "./Questions/Question3";
+
+export default function SurveyComponent() {
+  const [step, setStep] = useState(0);
+  const [answers, setAnswers] = useState({
+    question1: "",
+    question2: [] as string[],
+    question3: "",
+  });
+
+  const handleNext = () => setStep((prevStep) => prevStep + 1);
+  const handleBack = () => setStep((prevStep) => prevStep - 1);
+  const handleSubmit = () => {
+    handleNext();
+  };
+
+  const renderQuestion = () => {
+    switch (step) {
+      case 0:
+        return (
+          <Question1
+            answer={answers.question1}
+            setAnswer={(value) => setAnswers({ ...answers, question1: value })}
+          />
+        );
+      case 1:
+        return (
+          <Question2
+            answers={answers.question2}
+            setAnswers={(value) => setAnswers({ ...answers, question2: value })}
+          />
+        );
+      case 2:
+        return (
+          <Question3
+            answer={answers.question3}
+            setAnswer={(value) => setAnswers({ ...answers, question3: value })}
+          />
+        );
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <Box>
+      {step !== 3 && (
+        <>
+          <LinearProgress variant="determinate" value={(step / 3) * 100} />
+          <Box my={4}>{renderQuestion()}</Box>
+          <Box mt={4}>
+            {step > 0 && (
```

Comment:
what is step 0 and step 1, make constants with meaningful name of 0 and 1 steps.